src/garage/tf/q_functions/q_function.py

```diff
@@ -3,17 +3,9 @@
 
 
 class QFunction(abc.ABC):
-    """Q-function base class without Parameterzied.
-
-    Args:
-        name (str): Name of the Q-fucntion, also the variable scope.
-
-    """
-
-    def __init__(self, name):
-        self.name = name or type(self).__name__
-        self._variable_scope = None
+    """Q-function base class without Parameterzied."""
 
+    @abc.abstractmethod
     def get_qval_sym(self, *input_phs):
```

Comment:
Do we still need this function in the interface? Isn’t it equal to .build()?

Reply:
Yes, it is equal to build(). It seems that we can completely remove the QFunction interface? As the remaining method is clone().

Reply:
Sounds good to me